src/NavItem.js

```diff
@@ -28,7 +28,7 @@ class NavItem extends React.Component {
   }
 
   handleClick(e) {
-    if (this.props.onSelect) {
+    if (this.props.onSelect && this.props.onSelect.name !== 'bound handleCollapse') {
```

Comment:
my guess is that this will be too fragile across browsers and platforms, but it does illustrate the issue nicely